src/main/java/com/minecolonies/coremod/event/ClientEventHandler.java

```diff
@@ -77,18 +103,128 @@ public static void renderWorldLastEvent(@NotNull final RenderWorldLastEvent even
         }
+        else if (player.getHeldItemMainhand().getItem() == com.ldtteam.structurize.items.ModItems.buildTool)
+        {
+            handleRenderBuildTool(event, world, player);
+        }
+    }
+
+    /**
+     * Renders building bounding boxes into the client
+     *
+     * @param event  The caught event
+     * @param world  The world in which to render
+     * @param player The player for which to render
+     */
+    private static void handleRenderBuildTool(@NotNull final RenderWorldLastEvent event, final ClientWorld world, final PlayerEntity player)
+    {
+        final IColonyView colony = IColonyManager.getInstance().getClosestColonyView(world, player.getPosition());
+        if (colony == null)
+        {
+            return;
+        }
+
+        if (Settings.instance.getActiveStructure() == null)
+        {
+            return;
+        }
+
+        final BlockPos activePosition = Settings.instance.getPosition();
+        final Map<BlockPos, Triple<Blueprint, BlockPos, BlockPos>> newCache = new HashMap<>();
+        for (final IBuildingView buildingView : colony.getBuildings())
+        {
+            final BlockPos currentPosition = buildingView.getPosition();
+
+            if (activePosition.withinDistance(currentPosition, PREVIEW_RANGE))
+            {
+                if (blueprintCache.containsKey(currentPosition))
+                {
+                    newCache.put(currentPosition, blueprintCache.get(currentPosition));
+                }
+                else
+                {
+                    final StructureName sn =
+                      new StructureName(Structures.SCHEMATICS_PREFIX,
+                        buildingView.getStyle(),
+                        buildingView.getSchematicName() + buildingView.getBuildingMaxLevel());
+
+                    final String structureName = sn.toString();
+                    final String md5 = Structures.getMD5(structureName);
+
+                    final IStructureHandler wrapper = new LoadOnlyStructureHandler(world, buildingView.getID(), structureName, new PlacementSettings(), true);
+                    if (!wrapper.hasBluePrint() || !wrapper.isCorrectMD5(md5))
+                    {
+                        Log.getLogger().debug("Blueprint error, requesting" + structureName + " from server.");
+                        if (ServerLifecycleHooks.getCurrentServer() == null)
+                        {
+                            Network.getNetwork().sendToServer(new SchematicRequestMessage(structureName));
+                            continue;
+                        }
+                    }
+
+                    final Blueprint blueprint = wrapper.getBluePrint();
+
+                    if (blueprint != null)
+                    {
+                        Mirror mirror = buildingView.isMirrored() ? Mirror.FRONT_BACK : Mirror.NONE;
+
+                        final TileEntity tileEntity = world.getTileEntity(currentPosition);
```

Comment:
Could we get a highlight on the Tileentity to render aswell? would be very useful for finding buried buildings.
Should be a simple RenderBox() or so call, the code for that exists in structurize

Reply:
i'd say it'd be simpler to bring up the marker for the anchorPos (which is the TE in any building schematic)

Reply:
Highlighting the TE doesn't help with finding it, because if it's buried, the highlight will also be buried. By the time you find the highlight, you can see the hut anyway. I experimented with removing the depth check on the TE-highlight to make it shine through other solid blocks like a wallhack, but I couldn't make it work.
However, since the PR will render the full lvl5 building now (it works a bit different now than when you last saw it someaddons), it should be a bit easier to find the hut "intuitively".
Ashers, what do you mean with "bring up the marker"?

Reply:
https://github.com/ldtteam/Structurize/blob/9dfc02da242ee5b6f7ac4ecb49e6722f19ad262f/src/main/java/com/ldtteam/structurize/event/ClientEventSubscriber.java#L120

Reply:
Ah yes. Then my comment stands, the highlight isn't really useful for finding the hut, especially if it's buried. Would be a whole different story if the highlight were wallhacky, but no idea how to accomplish that.

Reply:
the tag hightling in structurize actually can be seen through blocks hmm

Reply:
oh funny its actually only the text highlightning which goes through blocks xD

Reply:
I tried reproducing the behavior of the icons of the workers, based on Ray's comment. It refused to work. The icons render as textures, and use `.depthTest(RenderState.DEPTH_ALWAYS)`, but no matter how much I fiddle around with this and culling, it just doesn't work. Maybe there's one or fifty other GL flags I need to set, I don't know.

Additionally, the icons of the workers are not visible through ghostblocks, so it wouldn't work for this PR anyway.

In short, sorry but I give up. Someone who understands rendering better than I do should take a look at this please, in a separate PR.

Reply:
Okay, we'll leave it for someone else to do it for a future PR. It's not actually so directly connected to this.